### DISCUSSION TITLE
Getting the build URL during the build?

### DISCUSSION THREAD

**Discussion by kutsurak** [buildbot/buildbot#6403] [Q&A] [answered]
Hi,

I wanted to ask if there is a way to get (or generate) the build URL during the build itself in order to pass it as a value to external processes? I saw that it is available to `MessageFormatter`, but unless I misunderstood the documentation the formatter (or components that use it) can be only used after the build completes. The build properties include the `buildnumber` and the `buildername`, but as far as I understand it I need a mapping from `buildername` to the builder number that is part of the URL. Is there some way to get that, or something else I could use that I have overlooked?

Thanks,
Panagiotis.

**Comment by vibbo** [comment, marked as answer]
Hi, would this help you?

https://github.com/buildbot/buildbot/blob/564802c0f69ae770f2dfb29faf3ac90b119c5959/master/docs/manual/configuration/properties.rst#url-for-build

**Reply by kutsurak** [reply]
This looks exactly like what I need. Thank you very much!
